pkg/genericclioptions/genericclioptions.go

```diff
@@ -23,13 +28,120 @@ type IOStreams struct {
 	ErrOut io.Writer
 }
 
-type ClientConfig struct {
-	Kubeconfig    string
+type sharedClientConfig struct {
```

Comment:
```suggestion
type ClientConfigBase struct {
```
I think this should be public so you can use it as an initializer in the struct that embeds it
(`shared` usually implies concurrent access)

Reply:
done